### ISSUE TITLE
predict result issue

### ISSUE THREAD

**Issue by konioyxgq** [Raschka-research-group/coral-cnn#15]
I used my own data set. First, the picture work with preprocess-cacd.py, and then using cacd-coral.py to get the age prediction. I tested 5 pictures and it didn't work well. But the results from the test pictures you provided are very good. Is this a problem with the generalization of the model? Or others?

**Comment by rasbt** [comment]
Yes it could be. The model wasn't trained for generalization across different datasets. To get better performance in that respect, one could combine all datasets into a single one and train a model based on that. Also see the related discussion here: https://github.com/Raschka-research-group/coral-cnn/issues/14

**Comment by konioyxgq** [comment]
I was testing on cacd-coral.py. I think the cacd dataset has about 160,000 data, which is a relatively large data set and should have good generalization capability. Do you think should do separate training in a specific use case?

**Comment by rasbt** [comment]
You may have to. I am not sure if `preprocess-cacd.py` works on different datasets -- haven't tested it. You have to make sure that your images look ok after preprocessing, after running your script. The reason we do preprocessing at all is that we wanted to compare our method directly to Niu et al's method, which did preprocessing as well.

**Comment by konioyxgq** [comment]
Well, thank you very much. In addition, I would like to ask why in the paper, only 14-62 age classification. Is it because there are very few samples in other age categories? For data balancing?

**Comment by rasbt** [comment]
>  I would like to ask why in the paper, only 14-62 age classification. Is it because there are very few samples in other age categories?

Yes. that's correct. I think there were only 1-10 examples for some classes.

**Comment by konioyxgq** [comment]
In fact, if I add edged the categories with a small number of other samples, would it be worse than the results in the article? I think it has almost no effect, or it doesn't have much impact. It's just that the category with a small sample has poor effect. Is that what I think? Will increasing the number of classifications have an impact on the outcome?

**Comment by rasbt** [comment]
I think you are right and it would have almost 0 impact because there are only so few examples for the edge cases. If you look at the appendix, we have also done experiments with an even narrow range of classes such that the dataset was balanced. The results are almost the same.

**Comment by konioyxgq** [comment]
Thank you very much for your patient response.
